tests/phpunit/elementor/includes/managers/test-controls.php

```diff
@@ -105,6 +105,72 @@ public function test_clear_stack_cache() {
 		$stack_cache_has_been_cleared = Plugin::$instance->controls_manager->has_stacks_cache_been_cleared();
 		$this->assertTrue( $stack_cache_has_been_cleared );
 
+		// Assert
+		$has_default_stack_cache_has_been_cleared = Plugin::$instance->controls_manager->has_stacks_cache_been_cleared( 'default' );
+		$this->assertTrue( $has_default_stack_cache_has_been_cleared );
+
+		// Assert
+		$stack_cache_has_been_cleared_none_existing_handle_id = Plugin::$instance->controls_manager->has_stacks_cache_been_cleared( 'none_existing_handle_id' );
+		$this->assertFalse( $stack_cache_has_been_cleared_none_existing_handle_id );
+
+		// Assert
+		$stacks = Plugin::$instance->controls_manager->get_stacks();
+		$this->assertEmpty( $stacks );
+	}
+
+	public function test_clear_stack_cache_with_handle_id() {
+		// Arrange
+		$handle_id = 'test_handle_id_post_1';
+
+		$control_data = [
+			'type'=>'select',
+			'tab'=>'style',
+			'section'=>'section_style',
+			'label'=>'Border Type',
+			'options'=>[''=>'Default'],
+			'none'=>'None',
+			'solid'=>'Solid',
+			'double'=>'Double',
+			'dotted'=>'Dotted',
+			'dashed'=>'Dashed',
+			'groove'=>'Groove',
+			'selectors'=>['{{WRAPPER}} .elementor-button'=>'border-style: {{VALUE}};'],
+			'separator'=>'before',
+			'classes'=>'elementor-group-control-border elementor-group-control elementor-group-control-border',
+			'name'=>'border_border',
+			'default'=>''
+		];
+
+		$widget = new Mock_Widget( [
+			'settings' => [],
+			'id' => '1',
+		], [] );
+
+		// Act
+		Plugin::$instance->controls_manager->open_stack( $widget );
+		Plugin::$instance->controls_manager->add_control_to_stack( $widget, 'border_border', $control_data, [] );
+
+		// Assert
+		$stacks = Plugin::$instance->controls_manager->get_stacks();
+		$this->assertNotEmpty( $stacks );
+
+		// Assert
+		$stack_cache_has_been_cleared = Plugin::$instance->controls_manager->has_stacks_cache_been_cleared( $handle_id );
+		$this->assertFalse( $stack_cache_has_been_cleared );
+
+		// Act
+		if ( ! $stack_cache_has_been_cleared ) {
+			Plugin::$instance->controls_manager->clear_stack_cache( $handle_id );
+		}
```

Comment:
It's already asserted as false.
```suggestion
		Plugin::$instance->controls_manager->clear_stack_cache( $handle_id );
```